jetty-core/jetty-http2/jetty-http2-tests/src/test/java/org/eclipse/jetty/http2/tests/IdleTimeoutTest.java

```diff
@@ -676,8 +678,15 @@ public boolean process(Request request, Response response, Callback callback)
                 System.err.println("processing request " + request.getHttpURI().getPath());
                 phaser.get().countDown();
                 // Hold the dispatched requests enough for the idle requests to idle timeout.
-                sleep(2 * idleTimeout);
-                callback.succeeded();
+                try
+                {
+                    Thread.sleep(2 * idleTimeout);
+                    callback.succeeded();
+                }
+                catch (InterruptedException e)
+                {
+                    callback.failed(e);
+                }
```

Comment:
It should not happen that this thread is interrupted... how it is the case?

Reply:
The test ends and reaches `@AfterEach void dispose()` which stops the server before the handler's `Thread.sleep()` call returns.

Reply:
Then maybe wait for the response is better?